### ISSUE TITLE
After 3 hours, the FastAPI is unable to respond and dies ECONNREFUSED

### ISSUE THREAD

**Issue by jeff-ixfintech** [tiangolo/fastapi#5578]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
FROM python:3.9-slim

RUN apt update && apt -y upgrade
RUN apt install -y curl

WORKDIR /app
COPY . .

RUN pip install poetry
RUN poetry config virtualenvs.create false
RUN poetry install --no-dev

EXPOSE 3000

CMD ["uvicorn", "pricefeed.test_websocket2:app", "--proxy-headers", "--host", "0.0.0.0", "--port", "3000", "--log-level", "trace"]
```

### Description

I deployed my FastAPI Docker Application with Uvicorn on AWS ECS Fargate with Network Load Balancer. 
The Health Check is `["CMD-SHELL", "curl -f http://127.0.0.1:3000/ws/status || exit 1"]`
However, after 3 hours of running the application, the fastapi endpoint is unreachable with the error ECONNREFUSED and the health check fails and forced the application restart. 

I found similar issues https://github.com/tiangolo/fastapi/issues/3700 , I have tried using gunicorn to run the application and still unable to solve the issue. Here is the command.
CMD ["gunicorn", "pricefeed.test_websocket:app", "--bind", "0.0.0.0:3000", "--worker-class", "uvicorn.workers.UvicornWorker"]

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.9.5

**Comment by jeff-ixfintech** [comment]
![screen2](https://user-images.githubusercontent.com/88023948/199648880-6be41739-958f-4f6c-b1d0-95b33054f671.png)
![screen3](https://user-images.githubusercontent.com/88023948/199648884-943e5aaf-5059-4d9a-be9f-fd3ffa70a0e4.png)
The above screenshots show that the container will restart after 170 minutes, i.e. it only lives for 170 minutes.

**Comment by jgould22** [comment]
You can probably try setting https://docs.gunicorn.org/en/stable/settings.html#max-requests to something like 1000 and see if that solves your problem in the short term and allow you some more time to debug

**Comment by tiangolo** [comment]
You can probably make a simple demo app as in the examples, with no processing or anything, and see if that simple app also dies after 170 minutes.

If it doesn't die, then it's probably something in your code. If it does, then it's something else, but try that first.

**Comment by jeff-ixfintech** [comment]
> 

Thanks. I will try this.

**Comment by jeff-ixfintech** [comment]
> You can probably try setting https://docs.gunicorn.org/en/stable/settings.html#max-requests to something like 1000 and see if that solves your problem in the short term and allow you some more time to debug

Thanks . I will try this out!

**Comment by jeff-ixfintech** [comment]
> You can probably make a simple demo app as in the examples, with no processing or anything, and see if that simple app also dies after 170 minutes.
> 
> If it doesn't die, then it's probably something in your code. If it does, then it's something else, but try that first.

Ok. I will try running the example app and get back to you afterwards. Thanks for your help!

**Comment by acidjunk** [comment]
Maybe offtopic: It might also be possible to run the stuff under AWS lambda; it depends a bit on your app though.

I am running some apps that way and it works really well: rather old POST describing the architecture:
https://www.renedohmen.nl/deploy-fastapi-on-amazon-serverless/

Feel free to ask for more info.

**Comment by jeff-ixfintech** [comment]
> Maybe offtopic: It might also be possible to run the stuff under AWS lambda; it depends a bit on your app though.
> 
> I am running some apps that way and it works really well: rather old POST describing the architecture: https://www.renedohmen.nl/deploy-fastapi-on-amazon-serverless/
> 
> Feel free to ask for more info.

Yes, but we are not just using the restapi. We have to implement the websocket features too. So lambda is not possible in my case. Thanks for your help!

**Comment by jeff-ixfintech** [comment]
> You can probably make a simple demo app as in the examples, with no processing or anything, and see if that simple app also dies after 170 minutes.
> 
> If it doesn't die, then it's probably something in your code. If it does, then it's something else, but try that first.

I tried running the demo app in AWS ECS and it has no problem at all. 
Finally, I just found out that the fastapi version in the project (we are using poetry) is not up-to-date, the version was 0.65.3, after i update it to 0.78.0, the problem seemed to be solved. 

But this is kinda weird as it is running normally in the local but not in ecs fargate. I am running the same image though.

Anyways, The problem is solved with updating the fastAPI version and other dependencies.

**Comment by tiangolo** [comment]
I'm glad to hear it's solved! In that case, you can close the issue. 🤓
